hr_expense_partner_config/models/hr_expense.py

```diff
@@ -0,0 +1,31 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = "hr.expense"
+
+    @api.onchange('product_id')
+    def _onchange_employee_config(self):
+        config = self.employee_id.expense_config_ids.filtered(lambda d: d.product_id == self.product_id)
+        if config:
+            self.payment_mode = config.payment_mode
+
+    @api.constrains('total_amount_currency')
+    def _check_daily_expense_limit(self):
+        for expense in self:
+            config = self.employee_id.expense_config_ids.filtered(lambda d: d.product_id == self.product_id and d.has_expense_limit)
+            if config:
+                same_expense_days = self.env['hr.expense'].search([
+                    ("date", "=", expense.date),
+                    ("product_id", "=", expense.product_id.id),
+                    ("employee_id", "=", expense.employee_id.id)
+                ])
+                total = 0
+                for day in same_expense_days:
+                    total += day.total_amount_currency
```

Comment:
Suggested compress code (not tested):
```suggestion
                total = sum([day.total_amount_currency for day in same_expense_days])
```

One question: `total_amount_currency` is amount in company currency (that is good), or original expense amount in custom expense currency?

Reply:
yes, it is the company's currency.